encoder/libav_encoder.cpp

```diff
@@ -173,17 +173,29 @@ void LibAvEncoder::initVideoCodec(VideoOptions const *options, StreamInfo const
 	// Apply general options.
 	encoderOptionsGeneral(options, codec_ctx_[Video]);
 
-	const char *format;
-	if (options_->output.empty() && options->libav_format.empty())
-		format = "h264";
-	else if (options->libav_format.empty())
-		format = nullptr;
+	const std::string tcp { "tcp://" };
+	const std::string udp { "udp://" };
+
+	// Setup an appropriate stream/container format.
+	const char *format = nullptr;
+	if (options->libav_format.empty())
+	{
+		if (options->output.empty() ||
+			options->output.find(tcp.c_str(), 0, tcp.length()) != std::string::npos ||
+			options->output.find(udp.c_str(), 0, udp.length()) != std::string::npos)
```

Comment:
This is slightly awkward to read, I find. Are we using C++20? Apparently that has a `starts_with`...

Reply:
Unfortunately we are using C++17, but starts_with is exactly what I want :(